manifests/pulp.pp

```diff
@@ -129,6 +142,52 @@
     }
   }
 
+  if $expose_mongodb {
+    include certs::mongodb
+    include certs::mongodb_client
+
+    Class['certs::mongodb_client'] ~> Class['pulp', 'apache::service']
+
+    $mongodb_bind_ip = $expose_mongodb_ipv6 ? {
+      true  => ['0.0.0.0', '::0'],
+      false => ['0.0.0.0'],
+    }
+
+    class { 'mongodb::server':
+      bind_ip => $mongodb_bind_ip,
+      ipv6    => $expose_mongodb_ipv6,
+      ssl     => true,
+      ssl_key => $certs::mongodb::mongodb_server_bundle,
+      ssl_ca  => $certs::mongodb::mongodb_server_ca_cert,
+      require => Class['certs::mongodb'],
+    }
+
+    $db_password = extlib::cache_data('foreman_cache_data', 'mongo_db_password', extlib::random_password(32))
+    $mongo_cmd = "mongo admin --quiet --host ${facts['networking']['fqdn']}:27017 --ssl --sslPEMKeyFile ${certs::mongodb_client::mongodb_client_bundle} --sslCAFile ${certs::mongodb_client::mongodb_client_ca_cert}"
+
+    exec { 'mongodb user':
+      command   => "${mongo_cmd} --eval \"db.createUser({user:'pulp',pwd:'${db_password}',roles:[{role:'dbOwner', db:'pulp_database'},{ role: 'readWrite', db: 'pulp_database'}]})\"",
+      path      => '/usr/bin:/bin',
+      logoutput => 'on_failure',
+      unless    => "${mongo_cmd} --eval 'printjson(db.system.users.find({\"_id\":\"pulp_database.pulp\"}).toArray().length)' | grep -q 1",
+      require   => CLass['mongodb::server'],
+    }
+
+    $mongodb_seeds_real = "${facts['networking']['fqdn']}:27017"
```

Comment:
This is essentially also used in $mongo_cmd. Could you define this earlier and reuse it there?